### PR TITLE
chore(ui): migrate Code component to TypeScript

### PR DESCRIPTION
**Pull request by gjaskiewicz**
# Summary

Migrate Code component to TypeScript

# Changes Made

- Migrate Code component to TypeScript

# Related Issues

- Issue 1: https://github.com/cloudoperators/juno/issues/219

# Testing Instructions

1. `npm i`
2. `npm run storybook`

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [ ] My changes generate no new warnings or errors.

### CODE REVIEW

**Review by andypf — CHANGES_REQUESTED** [review]
Good Job. 
In Storybook, the type of children is not displayed, whereas it was visible in the original version.

Screenshot 2024-11-15 at 08 51 30

**Comment by gjaskiewicz** [comment]
> > Good Job. In Storybook, the type of children is not displayed, whereas it was visible in the original version.
> > Screenshot 2024-11-15 at 08 51 30
> 
> Fixed it

image

**Review by andypf — APPROVED** [review]

File: packages/ui-components/src/components/Code/Code.component.tsx [hunk 1]
```diff
@@ -14,16 +13,16 @@ const codeStyles = `
 /** A basic inline <code> component.
  *   Accepts "content" prop or renders children as passed.
  */
-export const Code = ({ content = "", children = null, className = "", ...props }) => {
+export const Code: React.FC<CodeProps> = ({ content = "", children = null, className = "", ...props }) => {
   return (
     <code className={`juno-code ${codeStyles} ${className}`} {...props}>
-      {content || children}
+      {(content || children) as React.ReactNode}
```
Comment (guoda-puidokaite) [inline packages/ui-components/src/components/Code/Code.component.tsx:19]: I don't think we need to cast here? What do you think? 👀

**Review by guoda-puidokaite — CHANGES_REQUESTED** [review]
Nice one, almost there. 🚀Just one improvement.

File: packages/ui-components/src/components/Code/Code.component.tsx (hunk 1 above)
Comment (gjaskiewicz) [reply packages/ui-components/src/components/Code/Code.component.tsx]: Right

**Review by guoda-puidokaite — APPROVED** [review]
Looks great! 🚀

**Review by andypf — APPROVED** [review]

**Review by barsukov — APPROVED** [review]

**Pull request merged**